features/editing_documents.feature

```diff
@@ -0,0 +1,57 @@
+Feature: Editing documents for an application
+  Background:
+    Given I am logged in as an assessor
+    And a new planning application
+    And the planning application has a document with reference "FOOBAR"
+    And I view the planning application
+    And I view the document with reference "FOOBAR"
+
+  Scenario: I can replace the document
+    Given I attach a replacement file with path "spec/fixtures/images/proposed-roofplan.pdf"
+    And I fill in "Document reference(s)" with "DOC0001"
+    And I check "Floor"
+    And I check "Side"
+    And I choose "Yes" for "Do you want to list this document on the decision notice?"
+    And I choose "Yes" for "Should this document be made publicly available?"
+    When I press "Save"
+    Then the page contains "Document has been updated"
+    And the page contains "DOC0001"
+    And the page contains "Included in decision notice: Yes"
+    And the page contains "Public: Yes"
+
+  Scenario: I can edit the document's received at date
+    Given I fill in "Day" with "19"
+    And I fill in "Month" with "11"
+    And I fill in "Year" with "2021"
+    When I press "Save"
+    Then the page contains "Date received: 19 November 2021"
+
+  Scenario: I can see in the audit log when the received at date has been modified
+    Given I fill in "Day" with "19"
+    And I fill in "Month" with "11"
+    And I fill in "Year" with "2021"
+    And I press "Save"
+    And I view the document with reference "FOOBAR"
+    And I fill in "Day" with "22"
+    And I fill in "Month" with "11"
+    And I fill in "Year" with "2021"
+    When I press "Save"
+    Given I view the planning application audit
```

Comment:
There's a step for this: https://github.com/unboxed/bops/blob/master/features/step_definitions/auditing_steps.rb#L3